Just scan class files.  not MANIFEST.MF, LICENSE or any .java

**Desktop (please complete the following information):**
 - Vaadin Plugin version: 17.0.1

**Describe the bug**
ReflectionsClassFinder should only scan class files. at the moment there is to much useless log output

**To Reproduce**
.gradle clean build --debug

**Expected behavior**
just scan class files
```
org.reflections.ReflectionsException: could not create class object from file META-INF/MANIFEST.MF
        at org.reflections.scanners.AbstractScanner.scan(AbstractScanner.java:32)
        at org.reflections.Reflections.scan(Reflections.java:253)
        at org.reflections.Reflections.scan(Reflections.java:202)
        at org.reflections.Reflections.<init>(Reflections.java:123)
        at com.vaadin.gradle.ReflectionsClassFinder.<init>(ReflectionsClassFinder.kt:43)
        at com.vaadin.gradle.VaadinUtilsKt.getClassFinder(VaadinUtils.kt:63)
        at com.vaadin.gradle.VaadinUtilsKt.createNodeTasksBuilder(VaadinUtils.kt:157)
        at com.vaadin.gradle.VaadinPrepareFrontendTask.vaadinPrepareFrontend(VaadinPrepareFrontendTask.kt:89)
        at jdk.internal.reflect.GeneratedMethodAccessor624.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
...
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.reflections.ReflectionsException: could not create class file from MANIFEST.MF
        at org.reflections.adapters.JavassistAdapter.getOfCreateClassObject(JavassistAdapter.java:102)
        at org.reflections.adapters.JavassistAdapter.getOfCreateClassObject(JavassistAdapter.java:24)
        at org.reflections.scanners.AbstractScanner.scan(AbstractScanner.java:30)
        ... 108 more

```

--- a/build.gradle
+++ b/build.gradle
@@ -25,7 +25,7 @@ plugins {
 	id 'maven-publish'
     id 'idea'
     id 'com.gradle.plugin-publish' version '0.11.0'
-    id 'org.jetbrains.kotlin.jvm' version '1.4.20'
+    id 'org.jetbrains.kotlin.jvm' version '1.4.21'
 }
 
 /***********************************************************************************************************************
@@ -73,12 +73,12 @@ repositories {
 }
 
 dependencies {
-    compile("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.20")
+    compile("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.21")
     compile("com.vaadin:flow-migration:2.4.3")
     compile("org.zeroturnaround:zt-exec:1.12")
 
     testCompile("junit:junit:4.12")
-    testCompile("org.jetbrains.kotlin:kotlin-test:1.4.20")
+    testCompile("org.jetbrains.kotlin:kotlin-test:1.4.21")
 }
 
 idea {

--- a/src/functionalTest/kotlin/com/vaadin/gradle/AbstractGradleTest.kt
+++ b/src/functionalTest/kotlin/com/vaadin/gradle/AbstractGradleTest.kt
@@ -44,7 +44,7 @@ abstract class AbstractGradleTest {
         println("$testProjectDir/./gradlew ${args.joinToString(" ")}")
         val result: BuildResult = GradleRunner.create()
                 .withProjectDir(testProjectDir)
-                .withArguments(args.toList() + "--stacktrace" + "--info")
+                .withArguments(args.toList() + "--stacktrace" + "--debug") // use --debug to catch ReflectionsException: https://github.com/vaadin/vaadin-gradle-plugin/issues/99
                 .withPluginClasspath()
                 .withDebug(true)
                 .forwardOutput()   // a must, otherwise ./gradlew check freezes on windows!

--- a/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
+++ b/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
@@ -1,0 +1,72 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.gradle
+
+import com.vaadin.flow.server.frontend.scanner.ClassFinder
+import org.reflections.Reflections
+import org.reflections.util.ConfigurationBuilder
+import java.lang.annotation.Repeatable
+import java.lang.reflect.AnnotatedElement
+import java.net.URL
+import java.net.URLClassLoader
+
+/**
+ * A class finder using org.reflections.
+ *
+ * @param urls
+ *            the list of urls for finding classes.
+ *
+ * @since 2.0
+ */
+public class ReflectionsClassFinder(vararg urls: URL?) : ClassFinder {
+    @Transient
+    private val classLoader: ClassLoader
+
+    @Transient
+    private val reflections: Reflections
+
+    init {
+        classLoader = URLClassLoader(urls, null) // NOSONAR
+        reflections = Reflections(
+                ConfigurationBuilder().addClassLoader(classLoader)
+                        .setExpandSuperTypes(false).addUrls(*urls))
+    }
+
+    override fun getAnnotatedClasses(
+            clazz: Class<out Annotation?>): Set<Class<*>> {
+        val classes: MutableSet<Class<*>> = HashSet()
+        classes.addAll(reflections.getTypesAnnotatedWith(clazz, true))
+        classes.addAll(getAnnotatedByRepeatedAnnotation(clazz))
+        return classes
+    }
+
+    private fun getAnnotatedByRepeatedAnnotation(
+            annotationClass: AnnotatedElement): Set<Class<*>> {
+        val repeatableAnnotation = annotationClass
+                .getAnnotation(Repeatable::class.java)
+        return if (repeatableAnnotation != null) {
+            reflections.getTypesAnnotatedWith(
+                    repeatableAnnotation.value.java, true)
+        } else emptySet()
+    }
+
+    override fun getResource(name: String): URL? = classLoader.getResource(name)
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> loadClass(name: String): Class<T> = classLoader.loadClass(name) as Class<T>
+
+    override fun <T> getSubTypesOf(type: Class<T>): Set<Class<out T>> = reflections.getSubTypesOf(type)
+}

--- a/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
+++ b/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
@@ -40,9 +40,13 @@ public class ReflectionsClassFinder(vararg urls: URL?) : ClassFinder {
 
     init {
         classLoader = URLClassLoader(urls, null) // NOSONAR
-        reflections = Reflections(
-                ConfigurationBuilder().addClassLoader(classLoader)
-                        .setExpandSuperTypes(false).addUrls(*urls))
+        val cfg: ConfigurationBuilder = ConfigurationBuilder().apply {
+            addClassLoader(classLoader)
+            setExpandSuperTypes(false)
+            addUrls(*urls)
+            setInputsFilter { it!!.endsWith(".class") && !it.endsWith("module-info.class") } // only scan .class files: https://github.com/vaadin/vaadin-gradle-plugin/issues/99
+        }
+        reflections = Reflections(cfg)
     }
 
     override fun getAnnotatedClasses(

--- a/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
+++ b/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
@@ -50,19 +50,16 @@ public class ReflectionsClassFinder(vararg urls: URL) : ClassFinder {
     }
 
     override fun getAnnotatedClasses(clazz: Class<out Annotation>): Set<Class<*>> {
-        val classes: MutableSet<Class<*>> = HashSet()
+        val classes: MutableSet<Class<*>> = mutableSetOf()
         classes.addAll(reflections.getTypesAnnotatedWith(clazz, true))
         classes.addAll(getAnnotatedByRepeatedAnnotation(clazz))
         return classes
     }
 
     private fun getAnnotatedByRepeatedAnnotation(annotationClass: AnnotatedElement): Set<Class<*>> {
-        val repeatableAnnotation = annotationClass
-                .getAnnotation(Repeatable::class.java)
-        return if (repeatableAnnotation != null) {
-            reflections.getTypesAnnotatedWith(
-                    repeatableAnnotation.value.java, true)
-        } else emptySet()
+        val repeatableAnnotation: Repeatable = annotationClass.getAnnotation(Repeatable::class.java)
+            ?: return emptySet()
+        return reflections.getTypesAnnotatedWith(repeatableAnnotation.value.java, true)
     }
 
     override fun getResource(name: String): URL? = classLoader.getResource(name)

--- a/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
+++ b/src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt
@@ -31,7 +31,7 @@ import java.net.URLClassLoader
  *
  * @since 2.0
  */
-public class ReflectionsClassFinder(vararg urls: URL?) : ClassFinder {
+public class ReflectionsClassFinder(vararg urls: URL) : ClassFinder {
     @Transient
     private val classLoader: ClassLoader
 
@@ -49,16 +49,14 @@ public class ReflectionsClassFinder(vararg urls: URL?) : ClassFinder {
         reflections = Reflections(cfg)
     }
 
-    override fun getAnnotatedClasses(
-            clazz: Class<out Annotation?>): Set<Class<*>> {
+    override fun getAnnotatedClasses(clazz: Class<out Annotation>): Set<Class<*>> {
         val classes: MutableSet<Class<*>> = HashSet()
         classes.addAll(reflections.getTypesAnnotatedWith(clazz, true))
         classes.addAll(getAnnotatedByRepeatedAnnotation(clazz))
         return classes
     }
 
-    private fun getAnnotatedByRepeatedAnnotation(
-            annotationClass: AnnotatedElement): Set<Class<*>> {
+    private fun getAnnotatedByRepeatedAnnotation(annotationClass: AnnotatedElement): Set<Class<*>> {
         val repeatableAnnotation = annotationClass
                 .getAnnotation(Repeatable::class.java)
         return if (repeatableAnnotation != null) {

--- a/src/main/kotlin/com/vaadin/gradle/VaadinUtils.kt
+++ b/src/main/kotlin/com/vaadin/gradle/VaadinUtils.kt
@@ -19,7 +19,6 @@ import com.vaadin.flow.server.frontend.FrontendTools
 import com.vaadin.flow.server.frontend.FrontendUtils
 import com.vaadin.flow.server.frontend.NodeTasks
 import com.vaadin.flow.server.frontend.scanner.ClassFinder
-import com.vaadin.flow.server.scanner.ReflectionsClassFinder
 import elemental.json.JsonObject
 import elemental.json.impl.JsonUtil
 import org.gradle.api.Project

